### PR TITLE
fix: Keycloak deployments do not respect NodeSelector and Tolerations in the Argo CD CR(#450)

### LINKED ISSUE #450: ApplicationSets, Keycloak deployments does not respect NodeSelector and Tolerations in the Argo CD CR.
**Describe the bug**
#410  adds support to NodeSelectors and Tolerations. When NodeSelectors and Tolerations are added to the Argo CD CR as described in the #410 all the workloads gets scheduled on the expected Node. 

However, ApplicationSets and Keycloak are exceptions. Both ApplicationSets and Keycloak are not installed OOTB, A user needs to update them in the CR explicitly. When they are explicitly mentioned in the CR, they do not seem to respect the NodeSelector and Tolerations in the Argo CD CR.

**To Reproduce**
Run the operator locally using `make install && make run`
Create the Argo CD CR with spec as shown below.
```
spec:
  applicationSet: {}
  sso:
    provider: keycloak
  nodePlacement:
    nodeSelector:
      abc: abc
    tolerations:
      - effect: NoSchedule
        key: abc
        value: abc
```

**Expected behavior**
All the workloads including AppSets and Keycloak should get scheduled on the requested Node.

**Additional context**
NA

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind bug

Fixes #450 

**How to test changes / Special notes to the reviewer**:
#450

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
@saumeya Can you please review and test this PR ?

**Comment by saumeya** [comment]
LGTM @iam-veeramalla , tested and runs correctly!

File: controllers/argocd/keycloak.go [hunk 1]
```diff
@@ -279,6 +279,14 @@ func getKeycloakDeploymentConfigTemplate(cr *argoprojv1a1.ArgoCD) *appsv1.Deploy
 			},
 		},
 	}
+
+	if cr.Spec.NodePlacement != nil {
+		dc.Spec.Template.Spec.NodeSelector = cr.Spec.NodePlacement.NodeSelector
+		dc.Spec.Template.Spec.Tolerations = cr.Spec.NodePlacement.Tolerations
+	}
+
```
Comment (wtam2018) [inline controllers/argocd/keycloak.go:287]: If `cr.SpecNodePlacement` is nil, should we reset the dc.Spec?

**Comment by wtam2018** [comment]
Do we need to add unit /e2e tests?

File: controllers/argocd/keycloak.go (hunk 1 above)
Comment (iam-veeramalla) [reply controllers/argocd/keycloak.go:287]: That will be take care by the below section of code.

**Comment by iam-veeramalla** [comment]
> Do we need to add unit /e2e tests?

Sure, Added a Unit test.

**Review by wtam2018 — APPROVED** [review]
LGTM
Thanks @iam-veeramalla !

**Comment by jannfis** [comment]
Does this need to go into v1.3 branch?

**Comment by wtam2018** [comment]
It's been cherry-picked to v1.3.

**Pull request merged**